query_execution.go

```diff
@@ -195,6 +197,18 @@ func (q *queryExecution) executeChildStep(step *QueryPlanStep, boundaryIDs []str
 	return nil
 }
 
+func extractNonNilBoundaryResults(data []interface{}) []interface{} {
```

Comment:
is this worth extracting as a function?

Reply:
I just like having it extracted as it's pretty boilerplatey and I think the calling function reads better with it out.